hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -312,7 +316,7 @@ public void start(OzoneConfiguration configuration) {
       }
     }
 
-    if (snapshotDirectoryCleaningService == null &&
+    if (isSnapshotDeepCleaningEnabled && snapshotDirectoryCleaningService == null &&
```

Comment:
&& or 
|| ?

Reply:
It has to be && . Configuration to isSnapshotDeepCleaningEnabled should be enabled & also snapshotDirectoryCleaningService should not have been initialized before.